### PR TITLE
IEK changes

### PR DESCRIPTION
**Pull request by charvolant**
Allow common names with specific statuses to be pulled out of the common ruck and displayed in their own section. Tag common name search results with the language of the name.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,7 +18,21 @@ Customisations of code and i18n files should be made in the copy of `ala-bie` (n
...
+
+### Common Names Pull
+
+It is possible to "pull" common names with special status into their own section and have them displayed in a special way.
+To do this, use the following configuration settings:
+
+* `vernacularName.pull.active` Set to true for a pull display (*false* by default)
+* `vernacularName.pull.categories` A comma-separated list of status values that will cause the names to be pulled (*empty* by default)
+* `vernacularName.pull.label` The label for the pull section in the names tab (*Special Common Names* by default)
+* `vernacularName.pull.labelDetail` The detail to put into the label title
```
Comment (temi) [inline README.md]: default value missing

File: grails-app/taglib/au/org/ala/bie/BieTagLib.groovy [hunk 2]
```diff
@@ -185,31 +208,64 @@ class BieTagLib {
...
             out << text
         if (mark && defaultLocale.language != lang.language) {
-            String name = languageName(lang.language)
-            out << "&nbsp;<span class=\"annotation annotation-language\" title=\"${lang}\">${name}</span>"
+            def name = languages[lang.language] ?: [code: lang.toLanguageTag(), name: lang.displayName]
+            if (tag) {
+                out << "&nbsp;<span class=\"annotation annotation-language\" title=\"${name.code}\">${name.name}</span>"
+            } else {
+                String inLabel = message(code: 'label.in', default: 'in')
+                out << "<span class=\"in-marker\">&nbsp;${inLabel}&nbsp;</span><span class=\"language-name\" title=\"${lang}\">${name.name}</span>"
```
Comment (temi) [inline grails-app/taglib/au/org/ala/bie/BieTagLib.groovy]: `lang` in `title=\"${lang}\"` looks like an object. Is it correct?

File: grails-app/taglib/au/org/ala/bie/BieTagLib.groovy [hunk 3]
```diff
@@ -185,31 +208,64 @@ class BieTagLib {
...
+        Locale lang = buildLocale(attrs.lang)
+        boolean format = attrs.containsKey('format') ? attrs.format : true
+        def name = languages[lang.language]
+        def display = name?.name ?: attrs.default ?: lang.displayName
+        if (!format) {
+            out << display
+        } else {
+            out << "<span class=\"language-name\" title=\"${name?.code ?: lang.language}\">"
+            if (name.uri)
+                out << "<a href=\"${name.uri}\" onclick=\"window.open(this.href); return false;\">${display}</a>"
```
Comment (temi) [inline grails-app/taglib/au/org/ala/bie/BieTagLib.groovy]: Could use `target="_blank"` to open in new tab or window.

File: README.md (hunk 1 above)
Comment (charvolant) [reply README.md]: Fixed in https://github.com/AtlasOfLivingAustralia/bie-plugin/pull/170

File: grails-app/taglib/au/org/ala/bie/BieTagLib.groovy (hunk 2 above)
Comment (charvolant) [reply grails-app/taglib/au/org/ala/bie/BieTagLib.groovy]: Also fixed

File: grails-app/taglib/au/org/ala/bie/BieTagLib.groovy (hunk 3 above)
Comment (charvolant) [reply grails-app/taglib/au/org/ala/bie/BieTagLib.groovy]: Added

**Comment by charvolant** [comment]
Replaced by https://github.com/AtlasOfLivingAustralia/bie-plugin/pull/170

**Pull request closed without merging**